jill/download.py

```diff
@@ -144,11 +144,20 @@ def download_package(version=None, sys=None, arch=None, *,
     print(msg)
     registry = SourceRegistry(upstream=upstream)
     url = registry.query_download_url(version, system, architecture)
-    if not url:
-        msg = f"failed to find {release_str} in available upstream. Please try it later."
+    if not url and upstream != "Official":
```

Comment:
We still want to throw an error message when url is not found and `upstream == Official`, no?

Reply:
Yes, I believe the fallback workflow is:

1. try the specified upstream, e.g., `SJTUG`
2. if the URL for `SJTUG` doesn't exist, fall back to `Official` with a warning.
3. if the URL for `Official` doesn't exist, exit with an error.

Reply:
Also, the only possible trivial version that can not be found on all upstreams is "nightly" (this is how I tested the error handling), do we need to excluded it in the list of legitimate version strings?

Reply:
That might be too delicate, though. I'd prefer to not dealing with these content details in the script.

Reply:
For some extreme situations, if, say, `SJTUG` mirror stops syncing, and Official released 1.7.0. Then we still want to fall back to Official when users call `jill install 1.7.0 --upstream SJTUG`.